docarray/array/storage/base/getsetdel.py

```diff
@@ -126,6 +127,12 @@ def _del_docs_by_ids(self, ids):
         for _id in ids:
             self._del_doc_by_id(_id)
 
+    def _update_subindices_del(self, ids):
+        if getattr(self, '_subindices', None):
+            for selector, da in self._subindices.items():
+                ids_subindex = DocumentArray(self[ids])[selector, 'id']
+                da._del_docs_by_ids(ids_subindex)
```

Comment:
I think here should be `da._del_docs(ids_subindex)`, to also delete ids in self._offset2ids